frontend/js/tests/user/stats/UserArtistEvolutionActivity.test.tsx

```diff
@@ -0,0 +1,147 @@
+import * as React from "react";
+import { screen, waitFor, act } from "@testing-library/react";
+import { SetupServerApi, setupServer } from "msw/node";
+import { http, HttpResponse } from "msw";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import ArtistEvolutionActivityStreamGraph, {
+  UserArtistEvolutionActivityProps,
+} from "../../../src/user/stats/components/UserArtistEvolutionActivity";
+import * as userArtistEvolutionActivityResponse from "../../__mocks__/userArtistEvolutionActivity.json";
+import { renderWithProviders } from "../../test-utils/rtl-test-utils";
+
+const userProps: UserArtistEvolutionActivityProps = {
+  user: { name: "foobar" },
+  range: "week",
+};
+
+jest.mock("@nivo/stream", () => ({
+  ...jest.requireActual("@nivo/stream"),
+  ResponsiveStream: ({ children }: any) =>
+    children ? children({ width: 400, height: 400 }) : <div>Mock Stream Chart</div>,
+}));
+
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: { retry: false, staleTime: 0, gcTime: 0 },
+  },
+});
+
+const reactQueryWrapper = ({ children }: any) => (
+  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+);
+
+describe("ArtistEvolutionActivityStreamGraph", () => {
+  let server: SetupServerApi;
+
+  beforeAll(() => {
+    const handlers = [
+      // API now returns RAW rows; tests use that fixture
+      http.get("/1/stats/user/foobar/artist-evolution-activity", async ({ request }) => {
+        const url = new URL(request.url);
+        const range = url.searchParams.get("range");
+
+        switch (range) {
+          case "week":
+            return HttpResponse.json(userArtistEvolutionActivityResponse);
+          case "empty":
+            return HttpResponse.json({
+              payload: {
+                artist_evolution_activity: [],
+                range,
+                from_ts: 0,
+                to_ts: 0,
+                last_updated: 0,
+                user_id: "foobar",
+              },
+            });
+          case "month":
+            // Trigger error case
+            return HttpResponse.json({ message: "Failed to fetch data" }, { status: 500 });
+          default:
+            return HttpResponse.json(userArtistEvolutionActivityResponse);
+        }
+      }),
+    ];
+    server = setupServer(...handlers);
+    server.listen({ onUnhandledRequest: "error" });
+  });
+
+  beforeEach(() => {
+    queryClient.clear();
+  });
+
+  afterEach(() => {
+    queryClient.cancelQueries();
+    server.resetHandlers();
+  });
+
+  afterAll(() => {
+    server.close();
+  });
+
+  it("renders correctly", async () => {
+    await act(async () => {
+      renderWithProviders(
+        <ArtistEvolutionActivityStreamGraph {...userProps} />, {}, { wrapper: reactQueryWrapper }
+      );
+    });
+
+    await waitFor(() => {
+      expect(screen.getByTestId("artist-evolution")).toBeInTheDocument();
+    });
```

Comment:
For reference, I use this cheatsheet all the time. The grid at the top is useful to determine which type ofquery to use when you need to await or not:
https://testing-library.com/docs/react-testing-library/cheatsheet/#queries

```suggestion
    await screen.findByTestId("artist-evolution");
```